### DISCUSSION TITLE
ABI coding: representing bytes as Uint8Array

### DISCUSSION THREAD

**Discussion by timbrinded** [wevm/viem#927] [Idea / Feature Request]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Package Version

1.4.1

### Current Behavior

The `encodeFunctionData()` fn can't seem to turn uint arrays into bytes when using as an args parameter. 

Workaround is to use `fromBytes()` to first convert to a hex string, but this is non-ideal.

### Expected Behavior

`encodeFunctionData()` should be able to handle native bytes input

### Steps To Reproduce

```
import { encodeFunctionData } from 'viem';

const fragment = [
  {
    inputs: [{ internalType: 'bytes', name: 'innerCall', type: 'bytes' }],
    name: 'testFn',
    outputs: [],
    stateMutability: 'nonpayable',
    type: 'function',
  },
];

const notWorking = encodeFunctionData({
  abi: fragment,
  functionName: 'testFn',
  args: [new Uint8Array([0x01])],
});

console.log(notWorking + ' ❌');
```

### Link to Minimal Reproducible Example (StackBlitz, CodeSandbox, GitHub repo etc.)

https://stackblitz.com/edit/viem-getting-started-dmhpsz?file=main.ts

### Anything else?

Run the above with `pnpm example`

```

❯ pnpm example

> viem-example@0.0.0 example /home/projects/viem-getting-started-dmhpsz
> tsx main.ts

1 ✅
0x86fddd71000000000000000000000000000000000000000000000000000000000000002000000000000000000000000000000000000000000000000000000000000000010100000000000000000000000000000000000000000000000000000000000000 ✅
Error: hex_.replace is not a function
    at async ESMLoader.import (https://viemgettingstarteddmhpsz-q2xc.w-credentialless.staticblitz.com/blitz.bec01ace.js:35:1347715)
    at async i.loadESM (https://viemgettingstarteddmhpsz-q2xc.w-credentialless.staticblitz.com/blitz.bec01ace.js:35:352602)
    at async handleMainPromise (https://viemgettingstarteddmhpsz-q2xc.w-credentialless.staticblitz.com/blitz.bec01ace.js:35:1051734)
 ELIFECYCLE  Command failed with exit code 1.
```

**Comment by jxom** [comment]
Going to convert this one to a discussion as it’s not really a bug. viem represents `bytes(N)` as `Hex` in coding functions (with the escape hatch of `fromBytes` as you mentioned) as the performance cost of converting typed arrays (`Uint8Array`) to `Hex` internally is quite costly, and we don’t really want to encourage this.

If there is larger demand for this, maybe we can find an API to make it a bit easier.
